github_visualization/github_stats.py

```diff
@@ -0,0 +1,289 @@
+           Name of repository
+        '''
+        self.user_name = user_name
+        self.repositoty_name = repositoty_name
+
+        self.baseurl = "https://api.github.com/repos/%s/%s" % (
+            self.user_name, self.repositoty_name)
+
+    def get_total_contributions(self, weeks):
+        ''' Helper function to calculate total additions
+        and total deletions from a list of weeks of contributions
+        of a contributor
+
+        Parameters
+        -------------
+        weeks : list
+           List of weeks of contributions. Example:
+           [
+            {'w': 1254009600, 'a': 5, 'c': 2, 'd': 9},
+           ]
+
+        '''
+        total_additions = 0
+        total_deletions = 0
+        for week in weeks:
+            total_additions += week['a']
+            total_deletions += week['d']
+        return total_additions, total_deletions
+
+    def get_cumulative_contributors(self, contributors_list):
+        ''' Helper function to calculate total contributors as
+        new contributors join with time
+
+        Parameters
+        -------------
+        contributors_list : list
+           List of contributors with weeks of contributions. Example:
+           [
+               {
+                    'weeks': [
+                            {'w': 1254009600, 'a': 5, 'c': 2, 'd': 9},
+                        ],
+                    .....
+                },
+            ]
+
+        '''
+        contributorsJoinDate = {}
+
+        for contributor in contributors_list:
+            for week in contributor["weeks"]:
+                if(week["c"] > 0):
+                    joinDate = week['w']
+            if joinDate not in contributorsJoinDate:
+                contributorsJoinDate[joinDate] = 0
+            contributorsJoinDate[joinDate] += 1
+
+        cumulativeJoinDate = {}
+        cumulative = 0
+        for time in sorted(contributorsJoinDate):
+            print(time)
+            cumulative += contributorsJoinDate[time]
+            cumulativeJoinDate[time] = cumulative
+
+        cumulativeList = list(cumulativeJoinDate.items())
+        cumulativeList.sort()
+
+        return cumulativeList
+
+    def fetch_basic_stats(self):
+        ''' Fetch the basic stats
+
+        Returns
+        -------
+        basic_stats : dict
+            A dictionary containting basic statistics. For Example:
+            { 'response_code': 200,
+              'repo_name': 'dipy',
+              'repo_description': 'Diffusion MR Imaging in Python',
+              'html_url': 'https://github.com/nipy/dipy',
+              'subscribers': 41,
+              'forks': 142,
+              'forks_url': 'https://github.com/nipy/dipy/network'
+              'watchers': 94,
+              'open_issues': 154,
+              'is_private': False,
+              'stars': 94,
+              'stars_url': 'https://github.com/nipy/dipy/stargazers'
+            }
+
+        '''
+        basic_stats = {}
+        url = self.baseurl
```

Comment:
just to make all methods similar. Like in the next methods I append some string to the urls. Should I change it?